include/dxc/DXIL/DxilConstants.h

```diff
@@ -172,7 +172,10 @@ namespace DXIL {
     Invalid,
   };
 
+  // clang-format off
+  // Python lines need to be not formatted.
   /* <py::lines('SemanticKind-ENUM')>hctdb_instrhelp.get_enum_decl("SemanticKind", hide_val=True, sort_val=False)</py>*/
+  // clang-format on
```

Comment:
I don't think we want to format the generated code block either, unless we run clang-format as part of code generation.
Then, if we aren't clang-formatting code as part of code generation yet want the generated code to be formatted differently, we should update the code generator output formatting instead, right?

Reply:
I made the change minimal because I do want to enable formatting as part of the generation, but that requires people to all have clang-format available in the build environment, so I want to roll that out separately.

Reply:
Without excluding the generated portion, I imagine you would have to roll that out before doing the big reformatting change, otherwise you will get a build failure due to generated code formatting changes.  I'm not sure I feel good about including generated code for clang formatting and requiring clang-format to build the project.

Reply:
I don’t feel good about us checking generated code into the repo at all, but here we are.

My current process has been to clang-format everything then re-generate the files. That works fine, but does leave us with files that can’t be clang-formatted in their entirety.

Since the whole point of this is to require consistent code formatting for all incoming PRs, I don’t think it is inappropriate to require clang-format for building DXC. Given that Visual Studio and Xcode both ship clang format and it is available on every *nix platform I can think of I don’t think it is much of a burden to contribution.